src/common/Cargo.toml

```diff
@@ -158,3 +158,7 @@ path = "src/bin/default_config.rs"
 
 [lints]
 workspace = true
+
+[features]
+default = []
+disable_initial_exec_tls = ["tikv-jemalloc-ctl/disable_initial_exec_tls"]
```

Comment:
Note that this will be enabled in the binary level, so all crates will be affected. Maybe you can directly add it here if it's expected.

https://github.com/risingwavelabs/risingwave/blob/d2ec83c88b76e2a0d9b2c9ec59b05dd43edca780/Cargo.toml#L119

Reply:
disable_initial_exec_tls will only be enabled when build java binding, will it still affect rw binary?

Reply:
> will only be enabled when build java binding

This isn't true, as it's unified by hakari. You need to exclude `tikv-jemalloc-ctl` or `risingwave_java_binding` completely from hakari if you want to do so.

Reply:
I want to enable the `disable_initial_exec_tls` feature only when we build the Java dynamic library. In other cases, the default feature is ok.

Reply:
A little explanation: hakari tries to help you use the same feature sets regardless of what pkg you use in `cargo build -p <pkg>`, in order to avoid re-compilation. But you want exactly the reverse here: turn on different features for different pkgs.

Reply:
I think we should exclude risingwave_java_binding from hakari.

Reply:
I do think so, let me try. Not quite familiar with hakari before.

Reply:
By the way, I find `java_binding` does not utilize jemalloc at all but suffers from its issue. It might not be be a good idea for `common` to depend on `jemalloc`.

Reply:
> It might not be be a good idea for common to depend on jemalloc.

Agree. 

BTW, I just notice storage also depends on jemalloc, we might also need add the feature for storage.

Reply:
> It might not be be a good idea for common to depend on jemalloc.

Storage depends on jemalloc profiling, which has to depend on jemalloc. We may need to remove the profiling from compactor to address this.

Reply:
> > It might not be be a good idea for common to depend on jemalloc.
> 
> Storage depends on jemalloc profiling, which has to depend on jemalloc. We may need to remove the profiling from compactor to address this.

+1 to remove it from common (to `risingwave_common_heap_profiling`). https://github.com/risingwavelabs/risingwave/issues/11665

Reply:
Hmmm, but there's a `HeapProfilingConfig` in `common/config`, which seems hard to remove. 😅